Compiler warnings fixes for unused variable and incorrect type

cliu5764 opened the pull request:
Fixes for Compilation warnings/errors:
  include/zephyr/net/buf.h - casts assignment to prevent warning 
  include/zephyr/net/hostname.h - Just prevents a compiler warning due to unused arg. Should use "ARG_UNUSED(<arg>) rather than "(void) <arg>;"
  include/zephyr/net/net_context.h - casts and ARG_UNUSED() to prevent compiler warnings.
  include/zephyr/net/net_if.h - several ARG_UNUSED().
  include/zephyr/net/net_ip.h - cast in an assignment (presumably to prevent warning).
  include/zephyr/posix/time.h - cast to prevent warning
  include/zephyr/sys/byteorder.h - lots of casts to prevent warnings
  include/zephyr/sys/util.h - cast to prevent warnings


_edited by @ycsin: This PR is a re-incarnation of #69300_

ycsin commented on include/zephyr/net/buf.h:
```diff
@@ -869,7 +869,7 @@ struct net_buf_simple_state {
 				       struct net_buf_simple_state *state)
 {
-	state->offset = net_buf_simple_headroom(buf);
+        state->offset = (uint16_t)net_buf_simple_headroom(buf);
```
should use tab

cliu5764 replied:
Fixed

ycsin commented on include/zephyr/net/hostname.h:
```diff
@@ -103,6 +103,8 @@ int net_hostname_set_postfix(const uint8_t *hostname_postfix,
 static inline int net_hostname_set_postfix(const uint8_t *hostname_postfix,
 					   int postfix_len)
 {
+	(void) hostname_postfix;
```
should use `ARG_UNUSED` for these

ycsin replied:
This doesn't seem to be fixed

ycsin commented on include/zephyr/net/net_context.h:
```diff
@@ -435,7 +435,7 @@ static inline void net_context_set_accepting(struct net_context *context,
 		context->flags |= NET_CONTEXT_ACCEPTING_SOCK;
 	} else {
-		context->flags &= ~NET_CONTEXT_ACCEPTING_SOCK;
+	  context->flags &= (uint16_t)~NET_CONTEXT_ACCEPTING_SOCK;
```
should use tab for these

cliu5764 replied:
Fixed

cliu5764 replied:
[warnings.txt](https://github.com/zephyrproject-rtos/zephyr/files/14423686/warnings.txt)
Add the warnings generated from my build. If you want to reproduce the warning, try
git clone git@github.com:beechwoods-software/microROS-installer.git -b dave_patch_fix_with_pp_with_warnings ros_int; cd ros_int; ./docker_setup.sh
docker_setup.sh will set up a docker image and show you the build commands.

ycsin reviewed (changes requested):
what's the build command u used that generated build time warnings?

cfriedt commented on include/zephyr/net/net_context.h:
```diff
@@ -467,7 +467,7 @@ static inline void net_context_set_closing(struct net_context *context,
 		context->flags |= NET_CONTEXT_CLOSING_SOCK;
 	} else {
-		context->flags &= ~NET_CONTEXT_CLOSING_SOCK;
+	  context->flags &= (uint16_t)~NET_CONTEXT_CLOSING_SOCK;
```
This should be a tab as well

cfriedt commented on include/zephyr/net/net_context.h:
```diff
@@ -547,7 +547,7 @@ static inline void net_context_set_family(struct net_context *context,
 	    family == AF_PACKET || family == AF_CAN) {
 		/* Family is in BIT(4), BIT(5) and BIT(6) */
-		flag = family << 3;
+	  flag = (uint8_t)(family << 3);
```
Tab

cfriedt commented on include/zephyr/net/net_context.h:
```diff
@@ -589,7 +589,7 @@ static inline void net_context_set_type(struct net_context *context,
 	if (type == SOCK_DGRAM || type == SOCK_STREAM || type == SOCK_RAW) {
 		/* Type is in BIT(6) and BIT(7)*/
-		flag = type << 6;
+	  flag = (uint16_t)(type << 6);
```
Tab

jukkar commented:
This seems to be re-incarnation of #69300. There is normally no need to close and then re-create a new PR, you can always force push over the existing PR. This way the already written comments are not lost.

cfriedt commented on include/zephyr/posix/time.h:
```diff
@@ -83,7 +83,7 @@ extern "C" {
 static inline int32_t _ts_to_ms(const struct timespec *to)
 {
-	return (to->tv_sec * MSEC_PER_SEC) + (to->tv_nsec / NSEC_PER_MSEC);
+  return (int32_t)((to->tv_sec * MSEC_PER_SEC) + (to->tv_nsec / NSEC_PER_MSEC));
```
Maybe it's rendering incorrectly on mobile, but the indent does not look like a tab

cliu5764 replied:
Fixed

cfriedt commented on include/zephyr/sys/util.h:
```diff
@@ -532,7 +532,7 @@ size_t hex2bin(const char *hex, size_t hexlen, uint8_t *buf, size_t buflen);
 static inline uint8_t bcd2bin(uint8_t bcd)
 {
-	return ((10 * (bcd >> 4)) + (bcd & 0x0F));
+  return (uint8_t)((uint8_t)((uint8_t)10 * (uint8_t)(bcd >> 4)) + (uint8_t)(bcd & 0x0F));
```
Indent looks wrong. Rather than cast multiple times, can this just use a uint32_t internally that is cast when returned?

cliu5764 replied:
Fixed

ycsin commented:
Please have a look at the contribution guidelines (the bot message in your previous PR) - your commit is missing a valid title, message & sign-off and it will fail the compliance checks

cliu5764 commented:
I just submitted a commit with changes according to
https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines
Let me know if it needs more TLC :-)
Thanks
Chauncy

On Tue, Feb 27, 2024 at 9:18 PM Yong Cong Sin ***@***.***>
wrote:

> Please have a look at the contribution guidelines (the bot message in your
> previous PR) - your commit is missing a valid title, message & sign-off and
> it will fail the CI's compliance checks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zephyrproject-rtos/zephyr/pull/69440#issuecomment-1968066354>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7W4WZC5ZAIVL3A766JMPL3YV2HWLAVCNFSM6AAAAABDZPLX56VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNRYGA3DMMZVGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

cliu5764 commented:
I thought I have all spaces with tabs. I was wrong. I blamed expandtab option in vim :-). Should be fixed now. Thanks

cliu5764 commented:
I see no failures on "Build with Clang/LLVM / clang-build ...". Can someone explain why this didn't pass? Thanks

ycsin commented:
Looking at the error message:

```
Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details

 3d0d0295bec7496b54dec80cd231ac738ec28c6d: author email (LIU, CHAUNCY <chauncy@beechwoods.com>) needs to match one of the signed-off-by entries.
3d0d0295bec7496b54dec80cd231ac738ec28c6d: author email (LIU, CHAUNCY <chauncy@beechwoods.com>) does not follow the syntax: First Last <email>.
Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style for more details.
```

Make sure that your git config looks like the following:

```
$ git config --list
user.name=Chauncy Liu
user.email=chauncy@beechwoods.com
```

To match your commit signed-off:

```
Signed-off-by: Chauncy Liu <chauncy@beechwoods.com>
```

also, additional indentation issue:

```
 CODE_INDENT: code indent should use tabs where possible
File:include/zephyr/net/net_context.h
Line:1008
 LEADING_SPACE: please, no spaces at the start of a line
File:include/zephyr/net/net_context.h
Line:1008
```

ycsin commented on include/zephyr/net/net_context.h:
```diff
@@ -1004,6 +1005,10 @@ static inline int net_context_create_ipv6_new(struct net_context *context,
 					      const struct in6_addr *src,
 					      const struct in6_addr *dst)
 {
+        ARG_UNUSED(context);
```
if you are using vscode, the `editorconfig` extension would apply the right indentation for different file extensions automatically for you

cliu5764 commented:
Done git commit --amend -s

cfriedt commented:
@cliu5764 - is there any possibility that you could paste the warnings into this PR in a code block?

cliu5764 commented:
> @cliu5764 - is there any possibility that you could paste the warnings into this PR in a code block?

include/zephyr/net/buf.h:
```
In file included from /b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net      _l2.h:16,
                  from /b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net      _if.h:29:
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/buf.h: In function 'net_b      uf_simple_save':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/buf.h:868:25: warning: co      nversion from 'size_t' {aka 'unsigned int'} to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconve      rsion]
  868 |         state->offset = net_buf_simple_headroom(buf);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~
```
include/zephyr/net/hostname.h:
```
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/hostname.h: In function '      net_hostname_set_postfix':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/hostname.h:74:59: warning      : unused parameter 'hostname_postfix' [-Wunused-parameter]
   74 | static inline int net_hostname_set_postfix(const uint8_t *hostname_postfix,
      |                                            ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/hostname.h:75:48: warning      : unused parameter 'postfix_len' [-Wunused-parameter]
   75 |                                            int postfix_len)
      |                                            ~~~~^~~~~~~~~~~
```
include/zephyr/net/net_context.h:
```
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h: In functio      n 'net_context_set_accepting':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h:397:35: war      ning: conversion from 'long unsigned int' to 'uint16_t' {aka 'short unsigned int'} changes the value of '42949      66783' [-Wconversion]
  397 |                 context->flags &= ~NET_CONTEXT_ACCEPTING_SOCK;
      |                                   ^
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h: In functio      n 'net_context_set_closing':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h:429:35: war      ning: conversion from 'long unsigned int' to 'uint16_t' {aka 'short unsigned int'} changes the value of '42949      66271' [-Wconversion]
  429 |                 context->flags &= ~NET_CONTEXT_CLOSING_SOCK;
      |                                   ^
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h: In functio      n 'net_context_set_family':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h:509:24: war      ning: conversion from 'int' to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  509 |                 flag = family << 3;
      |                        ^~~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h: In functio      n 'net_context_set_iface':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h:669:26: war      ning: conversion from 'int' to 'int8_t' {aka 'signed char'} may change value [-Wconversion]
  669 |         context->iface = net_if_get_by_iface(iface);
      |                          ^~~~~~~~~~~~~~~~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h: In functio      n 'net_context_is_proxy_enabled':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h:713:69: war      ning: unused parameter 'context' [-Wunused-parameter]
  713 | static inline bool net_context_is_proxy_enabled(struct net_context *context)
      |                                                 ~~~~~~~~~~~~~~~~~~~~^~~~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h: In functio      n 'net_context_create_ipv6_new':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h:826:67: war      ning: unused parameter 'context' [-Wunused-parameter]
  826 | static inline int net_context_create_ipv6_new(struct net_context *context,
      |                                               ~~~~~~~~~~~~~~~~~~~~^~~~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h:827:63: war      ning: unused parameter 'pkt' [-Wunused-parameter]
  827 |                                               struct net_pkt *pkt,
      |                                               ~~~~~~~~~~~~~~~~^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h:828:70: war      ning: unused parameter 'src' [-Wunused-parameter]
  828 |                                               const struct in6_addr *src,
      |                                               ~~~~~~~~~~~~~~~~~~~~~~~^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_context.h:829:70: war      ning: unused parameter 'dst' [-Wunused-parameter]
  829 |                                               const struct in6_addr *dst)
      |                                               ~~~~~~~~~~~~~~~~~~~~~~~^~~
```
include/zephyr/net/net_if.h:
```
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h: In function 'ne      t_if_oper_state_set':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h:731:24: warning:       comparison is always true due to limited range of data type [-Wtype-limits]
  731 |         if (oper_state >= NET_IF_OPER_UNKNOWN && oper_state <= NET_IF_OPER_UP) {
      |                        ^~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h: In function 'ne      t_if_ipv6_set_base_reachable_time':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h:1643:71: warning      : unused parameter 'iface' [-Wunused-parameter]
 1643 | static inline void net_if_ipv6_set_base_reachable_time(struct net_if *iface,
      |                                                        ~~~~~~~~~~~~~~~^~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h:1644:65: warning      : unused parameter 'reachable_time' [-Wunused-parameter]
 1644 |                                                        uint32_t reachable_time)
      |                                                        ~~~~~~~~~^~~~~~~~~~~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h: In function 'ne      t_if_ipv6_get_reachable_time':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h:1664:70: warning      : unused parameter 'iface' [-Wunused-parameter]
 1664 | static inline uint32_t net_if_ipv6_get_reachable_time(struct net_if *iface)
      |                                                       ~~~~~~~~~~~~~~~^~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h: In function 'ne      t_if_ipv6_set_reachable_time':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h:1694:71: warning      : unused parameter 'ipv6' [-Wunused-parameter]
 1694 | static inline void net_if_ipv6_set_reachable_time(struct net_if_ipv6 *ipv6)
      |                                                   ~~~~~~~~~~~~~~~~~~~~^~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h: In function 'ne      t_if_ipv6_set_retrans_timer':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h:1711:65: warning      : unused parameter 'iface' [-Wunused-parameter]
 1711 | static inline void net_if_ipv6_set_retrans_timer(struct net_if *iface,
      |                                                  ~~~~~~~~~~~~~~~^~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h:1712:59: warning      : unused parameter 'retrans_timer' [-Wunused-parameter]
 1712 |                                                  uint32_t retrans_timer)
      |                                                  ~~~~~~~~~^~~~~~~~~~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h: In function 'ne      t_if_ipv6_get_retrans_timer':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_if.h:1732:69: warning      : unused parameter 'iface' [-Wunused-parameter]
 1732 | static inline uint32_t net_if_ipv6_get_retrans_timer(struct net_if *iface)
      |                                                      ~~~~~~~~~~~~~~~^~~~~
```
include/zephyr/net/net_ip.h
```
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_ip.h: In function 'net_ipv6_is_prefix':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/net/net_ip.h:686:16: warning: conversion from 'int' to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  686 |         mask = ((0xff << (8 - remain)) ^ 0xff) << remain;
      |                ^
```
include/zephyr/posix/time.h:
```
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/posix/time.h: In function '_t      s_to_ms':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/posix/time.h:82:44: warning:       conversion from 'time_t' {aka 'long long int'} to 'int32_t' {aka 'int'} may change value [-Wconversion]
   82 |         return (to->tv_sec * MSEC_PER_SEC) + (to->tv_nsec / NSEC_PER_MSEC);
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
include/zephyr/sys/byteorder.h:
```
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_be16':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:311:18: warning: conversion from 'int' to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  311 |         dst[0] = val >> 8;
      |                  ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:312:18: warning: conversion from 'uint16_t' {aka 'short unsigned int'} to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  312 |         dst[1] = val;
      |                  ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_be24':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:326:18: warning: conversion from 'uint32_t' {aka 'unsigned int'} to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  326 |         dst[0] = val >> 16;
      |                  ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:327:22: warning: conversion from 'uint32_t' {aka 'unsigned int'} to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
  327 |         sys_put_be16(val, &dst[1]);
      |                      ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_be32':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:341:26: warning: conversion from 'uint32_t' {aka 'unsigned int'} to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
  341 |         sys_put_be16(val >> 16, dst);
      |                      ~~~~^~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:342:22: warning: conversion from 'uint32_t' {aka 'unsigned int'} to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
  342 |         sys_put_be16(val, &dst[2]);
      |                      ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_be48':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:356:26: warning: conversion from 'uint64_t' {aka 'long long unsigned int'} to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
  356 |         sys_put_be16(val >> 32, dst);
      |                      ~~~~^~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:357:22: warning: conversion from 'uint64_t' {aka 'long long unsigned int'} to 'uint32_t' {aka 'unsigned int'} may change value [-Wconversion]
  357 |         sys_put_be32(val, &dst[2]);
      |                      ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_be64':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:371:26: warning: conversion from 'uint64_t' {aka 'long long unsigned int'} to 'uint32_t' {aka 'unsigned int'} may change value [-Wconversion]
  371 |         sys_put_be32(val >> 32, dst);
      |                      ~~~~^~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:372:22: warning: conversion from 'uint64_t' {aka 'long long unsigned int'} to 'uint32_t' {aka 'unsigned int'} may change value [-Wconversion]
  372 |         sys_put_be32(val, &dst[4]);
      |                      ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_le16':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:386:18: warning: conversion from 'uint16_t' {aka 'short unsigned int'} to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  386 |         dst[0] = val;
      |                  ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:387:18: warning: conversion from 'int' to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  387 |         dst[1] = val >> 8;
      |                  ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_le24':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:401:22: warning: conversion from 'uint32_t' {aka 'unsigned int'} to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
  401 |         sys_put_le16(val, dst);
      |                      ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:402:18: warning: conversion from 'uint32_t' {aka 'unsigned int'} to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  402 |         dst[2] = val >> 16;
      |                  ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_le32':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:416:22: warning: conversion from 'uint32_t' {aka 'unsigned int'} to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
  416 |         sys_put_le16(val, dst);
      |                      ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:417:26: warning: conversion from 'uint32_t' {aka 'unsigned int'} to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
  417 |         sys_put_le16(val >> 16, &dst[2]);
      |                      ~~~~^~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_le48':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:431:22: warning: conversion from 'uint64_t' {aka 'long long unsigned int'} to 'uint32_t' {aka 'unsigned int'} may change value [-Wconversion]
  431 |         sys_put_le32(val, dst);
      |                      ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:432:26: warning: conversion from 'uint64_t' {aka 'long long unsigned int'} to 'uint16_t' {aka 'short unsigned int'} may change value [-Wconversion]
  432 |         sys_put_le16(val >> 32, &dst[4]);
      |                      ~~~~^~~~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h: In function 'sys_put_le64':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:446:22: warning: conversion from 'uint64_t' {aka 'long long unsigned int'} to 'uint32_t' {aka 'unsigned int'} may change value [-Wconversion]
  446 |         sys_put_le32(val, dst);
      |                      ^~~
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/byteorder.h:447:26: warning: conversion from 'uint64_t' {aka 'long long unsigned int'} to 'uint32_t' {aka 'unsigned int'} may change value [-Wconversion]
  447 |         sys_put_le32(val >> 32, &dst[4]);
      |                      ~~~~^~~~~
```
include/zephyr/sys/util.h:
```
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/util.h: In function 'bcd2bin':
/b/mros/no_zephyr_patch/microros_ws/firmware/zephyrproject/zephyr/include/zephyr/sys/util.h:503:36: warning: conversion from 'int' to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  503 |         return  ((10 * (bcd >> 4)) + (bcd & 0x0F));
      |                 ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
```

jukkar commented:
No objections to the actual changes, but as I said in the original PR, please split the changes to logical components, all network related fixes to one commit, posix to one, and utility fixes to one commit inside this PR.
Also there is this CI reported compliance issue
```
Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details

 6d9c64186eb99fb0502c2f9c49bef87d7e995220: author email (LIU, CHAUNCY <chauncy@beechwoods.com>) does not follow the syntax: First Last <email>.
Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details

 1: UC2 Signed-off-by: must have a full name
```

cfriedt commented on include/zephyr/sys/util.h:
```diff
@@ -532,7 +532,7 @@ size_t hex2bin(const char *hex, size_t hexlen, uint8_t *buf, size_t buflen);
 static inline uint8_t bcd2bin(uint8_t bcd)
 {
-	return ((10 * (bcd >> 4)) + (bcd & 0x0F));
+	return (uint8_t)((uint8_t)((uint8_t)10 * (uint8_t)(bcd >> 4)) + (uint8_t)(bcd & 0x0F));
```
The casts in this function take something away from the clarity.

Would it make more sense to use another variable?
```suggestion
	uint32_t t = bcd;

	t = (10 * (t >> 4)) + (t & 0x0F);
  return (uint8_t)t;
```

cliu5764 replied:
Fixed

cliu5764 commented:
> No objections to the actual changes, but as I said in the original PR, please split the changes to logical components, all network related fixes to one commit, posix to one, and utility fixes to one commit inside this PR. Also there is this CI reported compliance issue
> 
> ```
> Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details
> 
>  6d9c64186eb99fb0502c2f9c49bef87d7e995220: author email (LIU, CHAUNCY <chauncy@beechwoods.com>) does not follow the syntax: First Last <email>.
> Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details
> 
>  1: UC2 Signed-off-by: must have a full name
> ```



> No objections to the actual changes, but as I said in the original PR, please split the changes to logical components, all network related fixes to one commit, posix to one, and utility fixes to one commit inside this PR. Also there is this CI reported compliance issue
> 
> ```
> Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details
> 
>  6d9c64186eb99fb0502c2f9c49bef87d7e995220: author email (LIU, CHAUNCY <chauncy@beechwoods.com>) does not follow the syntax: First Last <email>.
> Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details
> 
>  1: UC2 Signed-off-by: must have a full name
> ```

fixed the full name format. please check. All the changes are include header files and the changes are small. I can see the benefit of splitting the changes for big code changes. I think the changes are pretty straight-forward here.

jukkar commented:
> All the changes are include header files and the changes are small. I can see the benefit of splitting the changes for big code changes. I think the changes are pretty straight-forward here.

It is not about whether the change is small or big, the point is that if some commit needs to be reverted, it is best if the commit is only dealing with related changes. In this PR you are mixing different changes to one commit. Probably there is no need to revert this commit, but it is a good habit to take that into account in advance. So that is why I wanted to see changes separated to individual commits.

andyross commented:
Hm... a lot of these changes are introducing truncations.  Those are hard to review in isolation and can introduce bugs (even security bugs!) if the types don't actually match.  To pick just one from the reviews above:

```
- context->flags &= ~NET_CONTEXT_CLOSING_SOCK;
+ context->flags &= (uint16_t)~NET_CONTEXT_CLOSING_SOCK;
```

What happens if context->flags is a uint32_t?  Well, where we used to clearing just the one bit.  Now we'd be that one bit, **and also the whole top half of the word**.   There's no way to know at review time which is which, right?  Now imagine if those extra cleared bits had security impact.   To be clear, I'm **absolutely not** accusing you of introducing security bugs!  But at the same time, if you did want to sneak a vulnerability into an open source project, you'd probably hide it in a PR just like this one.   Basically I'm making a smell argument that changes like this are ones we shouldn't be in the habit of approving in big hygine PRs.

Can we maybe rework this to avoid hard-to-audit truncations?  In that example, you could define NET_CONTEXT_CLOSING_SOCK to evaluate to short.   Or introduce a "clear_flags_bit" API for contexts, etc...  Lots of options.

cliu5764 commented:
> what's the build command u used that generated build time warnings?
We see the warning during our micro ros instegration. You can build with
git clone git@github.com:beechwoods-software/microROS-installer.git mros
cd mros
./docker_setup.sh

docker_setup.sh will setup the docker image for building the installer. It also give instruction how to start the build.

cliu5764 commented:
Can someone point to me what are the requested changes needed to be addressed? Thanks

ycsin commented:
> Can someone point to me what are the requested changes needed to be addressed? Thanks

Just commenting as a volunter:

1. @jukkar is requesting the changes to be "_separated to individual commits_".
2. and @andyross is suggesting you to "_rework this to avoid hard-to-audit truncations_"

cliu5764 commented:
broken down to https://github.com/zephyrproject-rtos/zephyr/pull/70076, https://github.com/zephyrproject-rtos/zephyr/pull/70074 and https://github.com/zephyrproject-rtos/zephyr/pull/70072.

cfriedt commented:
> broken down to https://github.com/zephyrproject-rtos/zephyr/pull/70076, https://github.com/zephyrproject-rtos/zephyr/pull/70074 and https://github.com/zephyrproject-rtos/zephyr/pull/70072.

@cliu5764 - I think what Jukka requested was to have individual commits in one pull request.

Since these are all fixing warnings, it's fair to bundle the commits together. I would maybe suggest closing the other PRs since they duplicates this one.

cliu5764 committed:
include: Avoid compiler warning about unused variable and unsupported type

Use ARG_UNUSED on unused variable.
Cast variables to correct type to avoid compilation warnings.
Avoid the use of kernel BIT() which hardcoded 1UL

Signed-off-by: Chauncy Liu <chauncy@beechwoods.com>

cliu5764 commented:
> > broken down to #70076, #70074 and #70072.
> 
> @cliu5764 - I think what Jukka requested was to have individual commits in one pull request.
> 
> Since these are all fixing warnings, it's fair to bundle the commits together. I would maybe suggest closing the other PRs since they duplicate this one.



> > broken down to #70076, #70074 and #70072.
> 
> @cliu5764 - I think what Jukka requested was to have individual commits in one pull request.
> 
> Since these are all fixing warnings, it's fair to bundle the commits together. I would maybe suggest closing the other PRs since they duplicate this one.


> Hm... a lot of these changes are introducing truncations. Those are hard to review in isolation and can introduce bugs (even security bugs!) if the types don't actually match. To pick just one from the reviews above:
> 
> ```
> - context->flags &= ~NET_CONTEXT_CLOSING_SOCK;
> + context->flags &= (uint16_t)~NET_CONTEXT_CLOSING_SOCK;
> ```
> 
> What happens if context->flags is a uint32_t? Well, where we used to clearing just the one bit. Now we'd be that one bit, **and also the whole top half of the word**. There's no way to know at review time which is which, right? Now imagine if those extra cleared bits had security impact. To be clear, I'm **absolutely not** accusing you of introducing security bugs! But at the same time, if you did want to sneak a vulnerability into an open source project, you'd probably hide it in a PR just like this one. Basically I'm making a smell argument that changes like this are ones we shouldn't be in the habit of approving in big hygine PRs.
> 
> Can we maybe rework this to avoid hard-to-audit truncations? In that example, you could define NET_CONTEXT_CLOSING_SOCK to evaluate to short. Or introduce a "clear_flags_bit" API for contexts, etc... Lots of options.



> Hm... a lot of these changes are introducing truncations. Those are hard to review in isolation and can introduce bugs (even security bugs!) if the types don't actually match. To pick just one from the reviews above:
> 
> ```
> - context->flags &= ~NET_CONTEXT_CLOSING_SOCK;
> + context->flags &= (uint16_t)~NET_CONTEXT_CLOSING_SOCK;
> ```
> 
> What happens if context->flags is a uint32_t? Well, where we used to clearing just the one bit. Now we'd be that one bit, **and also the whole top half of the word**. There's no way to know at review time which is which, right? Now imagine if those extra cleared bits had security impact. To be clear, I'm **absolutely not** accusing you of introducing security bugs! But at the same time, if you did want to sneak a vulnerability into an open source project, you'd probably hide it in a PR just like this one. Basically I'm making a smell argument that changes like this are ones we shouldn't be in the habit of approving in big hygine PRs.
> 
> Can we maybe rework this to avoid hard-to-audit truncations? In that example, you could define NET_CONTEXT_CLOSING_SOCK to evaluate to short. Or introduce a "clear_flags_bit" API for contexts, etc... Lots of options.

I just pushed a new commit which reverts the casting of (uint16_t). I fixed by change the define from BIT(9) to (1<<9). The kernel BIT() tends to use hardcoded UL. I will let the compiler to do the right type conversion. Can you review it? Thanks

cliu5764 commented:
@ycsin I am still seeing one requested change from you. I answered it in https://github.com/zephyrproject-rtos/zephyr/pull/69440#issuecomment-1984240914. Can we resolve this? Thanks

cfriedt commented on include/zephyr/net/net_context.h:
```diff
@@ -60,10 +60,10 @@ enum net_context_state {
 
 /** Is the socket accepting connections */
-#define NET_CONTEXT_ACCEPTING_SOCK  BIT(9)
+#define NET_CONTEXT_ACCEPTING_SOCK  (1<<9)
```
This should probably be reverted

cfriedt commented on include/zephyr/net/net_context.h:
```diff
@@ -60,10 +60,10 @@ enum net_context_state {
 
 /** Is the socket closing / closed */
-#define NET_CONTEXT_CLOSING_SOCK  BIT(10)
+#define NET_CONTEXT_CLOSING_SOCK  (1<<10)
```
This should probably be reverted

cfriedt commented:
> what's the build command u used that generated build time warnings?

@cliu5764 - if you're unable to provide the build command, I would suggest maybe simply adding `-Wno-conversion` to your `CFLAGS`. That would eliminate the entire category of warnings that you have reported.

With that, this change should be more or less equivalent to adding `ARG_UNUSED()` in a few places, which should be considerably easier to review and merge.

cfriedt reviewed (changes requested):
> what's the build command u used that generated build time warnings?

@cliu5764 - if you're unable to provide the build command, I would suggest maybe simply adding `-Wno-conversion` to your `CFLAGS`. That would eliminate the entire category of warnings that you have reported.

With that, this change should be more or less equivalent to adding `ARG_UNUSED()` in a few places, which should be considerably easier to review and merge.
